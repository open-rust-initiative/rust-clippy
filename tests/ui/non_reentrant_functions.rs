#![warn(clippy::non_reentrant_functions)]
#![allow(unused)]
#![feature(rustc_private)]
extern crate libc;

use std::ffi::{CStr, CString};

fn test_libc_localtime() {
    // test code goes here
    unsafe {
        let _tm = libc::localtime(&0i64 as *const libc::time_t);
    }
}

fn test_libc_strtok() {
    let string = CString::new("welcome-to-rust").unwrap();
    let string = string.as_ptr() as *mut libc::c_char;
    let delim = CString::new(" - ").unwrap();
    let delim = delim.as_ptr();

    let mut token = unsafe { libc::strtok(string, delim) };
    while !token.is_null() {
        println!("{:?}", unsafe { CStr::from_ptr(token) });
        token = unsafe { libc::strtok(std::ptr::null_mut(), delim) };
    }
}

fn test_locatime() {
    fn localtime() {}
    localtime();
}

fn test_strtok() {
    fn strtok() {}
    strtok();
}

fn main() {}
