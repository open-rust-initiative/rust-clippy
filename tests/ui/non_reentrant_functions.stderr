error: consider using the reentrant version of the function
  --> $DIR/non_reentrant_functions.rs:11:19
   |
LL |         let _tm = libc::localtime(&0i64 as *const libc::time_t);
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::non-reentrant-functions` implied by `-D warnings`

error: consider using the reentrant version of the function
  --> $DIR/non_reentrant_functions.rs:21:30
   |
LL |     let mut token = unsafe { libc::strtok(string, delim) };
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: consider using the reentrant version of the function
  --> $DIR/non_reentrant_functions.rs:24:26
   |
LL |         token = unsafe { libc::strtok(std::ptr::null_mut(), delim) };
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

